### ISSUE TITLE
DDR: Fix foot detection on 64-bit p4io

### ISSUE THREAD

**Issue by TheDDREnjoyer** [spice2x/spice2x.github.io#150]
## Description of change
My previous patch that added p4io support hardcoded the timer source that the game was using to determine tick time: `timeGetTime()`.  This patch swaps that call out with a shim that will attempt to get the timestamp from `arkGetTickTime64()` in `ark*.dll`. If for whatever reason `arkGetTickTime64` is not present in the game dll, it will fall back to `timeGetTime`.

This patch maybe? also fixes step detection in bio2, if that was a problem before? I don't have data that works properly with it so I can't test it, but looking around in IDA it seems like it has the same issue as p4io (p3io inserts the required timestamp, p4io needs it to come from the buffer returned by libacio, and bio2 looks extremely similar). If anyone tries it out and it works let me know I'd be very interested to hear. Also, my email is in the patch file if you ever want to chat outside of GH.

## Testing
MDX was booted with 64-bit versions of `arkmdxp3.dll` and `arkmdxp4.dll` and the tutorial was played to ensure that steps could be detected in both architectures.

## Patch file
git format-patch
[0001-ddr-Fix-foot-detection-on-64-bit-p4io.patch](https://github.com/spice2x/spice2x.github.io/files/14625925/0001-ddr-Fix-foot-detection-on-64-bit-p4io.patch)

**Comment by LupinThidr** [comment]
On bio2 the arrow recepticle(?) would move when arrows were stepped, but actual notes would just fall past as if nothing was being hit, is that what you were referring to by "foot detection"?
64-bit modules for A20+ (including arkmdxbio2.dll) are supposedly public now.

**Comment by TheDDREnjoyer** [comment]
> On bio2 the arrow recepticle(?) would move when arrows were stepped, but actual notes would just fall past as if nothing was being hit, is that what you were referring to by "foot detection"? 64-bit modules for A20+ (including arkmdxbio2.dll) are supposedly public now.

yeah, that's exactly the same issue that p4io was having because of the missing timestamp
